### DISCUSSION TITLE
Can I use Password grant with Keycloak?

### DISCUSSION THREAD

**Discussion by nrsina** [dexidp/dex#2340] [Q&A] [answered]
I'm trying to understand how can I connect DEX to Keycloak and use the Resource Owner Password Grant type.
As I understood, DEX now supports password grant. I don't want to use redirects like in Authorization Code flow.
Any help would be appreciated.

**Comment by nabokihms** [comment, marked as answer]
You can only use password grant with several connectors - local, LDAP, OpenStack Keystone, Atlassian Crowd. 

These connectors are special because we use a Dex login form to log in with them. In this case, Dex knows the user login and password.

For connectors such as OIDC, Dex doesn't know anything about user credentials. It only receives the info about user authentication with a callback.

**Reply by nrsina** [reply]
Thank you for your answer.
